packages/lib-utils/src/components/status/StatusBox.tsx

```diff
@@ -0,0 +1,99 @@
+import { Alert, EmptyState, EmptyStateBody, Spinner, Text, Title } from '@patternfly/react-core';
+import * as _ from 'lodash-es';
+import * as React from 'react';
+
+export type LoadError = {
+  message: string;
+  status: number;
+};
+
+type AccessDeniedProps = {
+  message: string | React.ComponentType;
+};
+
+export const AccessDenied: React.FC<AccessDeniedProps> = ({
+  message = 'Insufficient access rights.',
+}) => (
+  <EmptyState>
+    <Title headingLevel="h2">Restricted Access</Title>
+    <EmptyStateBody>
+      <Text>You don&apos;t have access to this section due to cluster policy.</Text>
+      {message && (
+        <Alert isInline variant="danger" title="Error details">
+          {message}
+        </Alert>
+      )}
+    </EmptyStateBody>
+  </EmptyState>
+);
+
+export type StatusBoxProps<D> = {
+  areFiltersApplied?: boolean;
+  data?: D[];
+  children?: React.ReactElement;
+  emptyLabel?: string;
+  loadError?: LoadError;
+  loaded?: boolean;
+  LoadErrorDefaultMsg?: React.ComponentType<Record<string, never>>;
```

Comment:
`Record<string, never>` infers into an empty object, see [here](https://github.com/typescript-eslint/typescript-eslint/issues/2063#issuecomment-675156492) for details.

I'd suggest adding the following into `packages/common/src/types/common.ts` right after `AnyObject` type declaration:
```ts
/**
 * Use the `EmptyObject` type for objects that have no properties.
 * 
 * @see https://github.com/typescript-eslint/typescript-eslint/issues/2063#issuecomment-675156492
 */
export type EmptyObject = Record<string, never>;
```

and using `EmptyObject` wherever it's needed.